codegen-core/common-test-models/constraints.smithy

```diff
@@ -193,6 +200,9 @@ structure ConstrainedHttpBoundShapesOperationInputOutput {
     @httpHeader("X-Range-Integer")
     rangeIntegerHeader: RangeInteger,
 
+    @httpHeader("X-Range-Short")
+    rangeShortHeader: RangeShort,
```

Comment:
The other HTTP bindings which are compatible with short are [@httpLabel](https://smithy.io/2.0/spec/http-bindings.html#httplabel-trait), and [@httpResponseCode](https://smithy.io/2.0/spec/http-bindings.html#httpresponsecode-trait). Do we want to also add coverage for them?

Reply:
Yes, that's [here](https://github.com/awslabs/smithy-rs/pull/2034/files#diff-d1ac0e3b7b50015c41afa00ec53ce786a3e1744483220a0afab4c1c1a0d7503fR187)

Reply:
Turns out I'm blind